lib/action_cable/connection/base.rb

```diff
@@ -165,6 +165,28 @@ def on_close
         end
 
 
+        def allow_request_origin?
+          return true if server.config.disable_request_forgery_protection
+
+          if env['HTTP_ORIGIN'].present?
+            origin_host = URI.parse(env['HTTP_ORIGIN']).host
```

Comment:
The Origin header is an opaque identifier, so let's do literal comparisons rather than parsing. Whitelist actual Origin values rather than hosts only.

That's important because the URL scheme is included in the Origin as well as, in most browsers, the port number.